packages/bot-web-ui/src/components/chart/__tests__/toolbar-widgets.spec.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import { isDesktop, isMobile } from '@deriv/shared';
+import { mockStore, StoreProvider } from '@deriv/stores';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { act, render, screen } from '@testing-library/react';
+import { mock_ws } from 'Utils/mock';
+import { DBotStoreProvider, mockDBotStore } from 'Stores/useDBotStore';
+import ToolbarWidgets from '../toolbar-widgets';
+
+jest.mock('@deriv/shared', () => ({
+    ...jest.requireActual('@deriv/shared'),
+    isMobile: jest.fn(),
+    isDesktop: jest.fn(),
+}));
+
+jest.mock('@deriv/bot-skeleton/src/scratch/blockly', () => jest.fn());
+jest.mock('@deriv/bot-skeleton/src/scratch/dbot', () => ({
+    saveRecentWorkspace: jest.fn(),
+    unHighlightAllBlocks: jest.fn(),
+}));
+jest.mock('@deriv/bot-skeleton/src/scratch/hooks/block_svg', () => jest.fn());
+
+// // Mocking the imports from @deriv/deriv-charts
+jest.mock('@deriv/deriv-charts', () => ({
+    ...jest.requireActual('@deriv/deriv-charts'),
+    __esModule: true,
+    ChartMode: jest.fn(() => <div>Mocked ChartMode</div>),
+    DrawTools: jest.fn(() => <div>Mocked DrawTools</div>),
+    Share: jest.fn(() => <div>Mocked Share</div>),
+    StudyLegend: jest.fn(() => <div>Mocked StudyLegend</div>),
+    ToolbarWidget: jest.fn(({ children }) => <div>{children}</div>),
+    Views: jest.fn(() => <div>Mocked Views</div>),
+}));
+
+describe('ToolbarWidgets', () => {
+    let wrapper: ({ children }: { children: JSX.Element }) => JSX.Element;
+
+    const mockUpdateChartType = jest.fn();
+    const mockUpdateGranularity = jest.fn();
+
+    beforeAll(() => {
+        const mock_store = mockStore({});
+        const mock_DBot_store = mockDBotStore(mock_store, mock_ws);
+
+        wrapper = ({ children }: { children: JSX.Element }) => (
+            <StoreProvider store={mock_store}>
+                <DBotStoreProvider ws={mock_ws} mock={mock_DBot_store}>
+                    {children}
+                </DBotStoreProvider>
+            </StoreProvider>
+        );
+    });
+
+    it('should render ToolbarWidgets in desktop', () => {
+        act(() => {
```

Comment:
do we need act here ?

Reply:
@vinu-deriv Bhai it's actually not a hard requirement but I am trying to follow a practice as per the document 
"Act: To prepare a component for assertions, wrap the code render it and perform updates inside an act() call. This makes your test run closer to how React works in the browser." So even though the test will still run without act here but as I am updating a state value here (mobile/desktop) so I am putting it inside act. But again it's not mandatory and if needed we can remove it.